railties/lib/rails/application.rb

```diff
@@ -438,8 +438,12 @@ def secret_key_base
     # Decrypts the credentials hash as kept in +config/credentials.yml.enc+. This file is encrypted with
     # the Rails master key, which is either taken from <tt>ENV["RAILS_MASTER_KEY"]</tt> or from loading
     # +config/master.key+.
+    # If specific credentials file exists for current environment, it takes precedence, thus for +production+
+    # environment look first for +config/credentials/production.yml.enc+ with master key taken
+    # from <tt>ENV["RAILS_MASTER_KEY"]</tt> or from loading +config/credentials/production.key+.
+    # Default behavior can be overwritten by setting +config.credentials.content_path+ and +config.credentials.key_path+.
     def credentials
-      @credentials ||= encrypted("config/credentials.yml.enc")
+      @credentials ||= encrypted(config.credentials.content_path, key_path: config.credentials.key_path)
```

Comment:
Is there an issue here where this line is ignoring `env_key` ?

Reply:
Although `RAILS_#{options[:environment].upcase}_KEY` was originally supported, it was removed in #33928.  Use `RAILS_MASTER_KEY` instead.

Reply:
Perfect, I see how to use it now thanks